lisa/tools/fdisk.py

```diff
@@ -51,16 +51,32 @@ def make_partition(self, disk_name: str, file_system: FileSystem) -> None:
         else:
             raise LisaException(f"Unrecognized file system {file_system}.")
 
-    def delete_partition(self, disk_name: str) -> None:
+    def delete_partitions(self, disk_name: str) -> None:
         """
-        disk: delete one partition against the disk.
-        Delete the only partition of this disk.
+        disk: delete all partitions against the disk.
+        Get the partitions count from /proc/partitions.
+        Delete all partitions of this disk.
         """
-        # d => delete a partition
-        # w => write table to disk and exit
-        self.node.execute(
-            f"(echo d; echo w) | {self.command} {disk_name}", shell=True, sudo=True
+        # /dev/sdc => sdc
+        disk_label = disk_name.split("/")[-1]
+        # /dev/sdc
+        # /dev/sdc1
+        # /dev/sdc2
+        partitions = self.node.execute(
```

Comment:
Avoid to use grep in command line. Read all from cat, and parse it in Python. It can show more raw output from commands, and easy to troubleshooting.

Reply:
fixed